### DISCUSSION TITLE
API call to get ip-addresses

### DISCUSSION THREAD

**Discussion by maekee** [netbox-community/netbox#7874] [Q&A] [answered]
Hello,
We have a few thousand ip-addresses and i am trying to loop the vlans/ranges/prefixes and get ip-addresses connected to these. But i cant find a relationship to query against.

Thinking something like /api/ipam/ip-addresses?range=x or /api/ipam/ip-addresses?prefix=x

See on the api docs under ip-addresses that there are a q field.
Is this for Complex Lookups? (https://docs.djangoproject.com/en/3.2/topics/db/queries/#complex-lookups-with-q-objects)

Suggestions?

**Comment by markkuleinio** [comment, marked as answer]
Hi, there is `/api/ipam/ip-addresses/?parent=x.x.x.x/y`

**Reply by maekee** [reply]
Thanks alot, just what i was looking for. Didnt really know what data to use on parent.. tried id

**Reply by APIfriendly** [reply]
Thanks again, i also was looking for the same thing as i didn't found the API route for the GUI view https://netbox/ipam/prefixes/xx/ip-addresses/
